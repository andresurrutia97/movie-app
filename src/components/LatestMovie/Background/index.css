.latest-movie__img {
  filter: blur(1px);
  opacity: 0.6;
  margin-top: 64px;
}

@media (min-width: 768px) {
  .latest-movie__img {
    margin-top: 0;
  }
}
