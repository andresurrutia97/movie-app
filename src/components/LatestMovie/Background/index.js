import React from 'react';
import PropTypes from 'prop-types';

import './index.css';
import { getImageUrl } from '../../../utils/urls';

const ImgBackground = ({ img }) => {
  const bgImg = img ? (
    <img className="w-full latest-movie__img" src={getImageUrl(img)} alt="" />
  ) : (
    ''
  );

  return <>{bgImg}</>;
};

ImgBackground.propTypes = {
  img: PropTypes.string,
};

ImgBackground.defaultProps = {
  img: '',
};

export default ImgBackground;
