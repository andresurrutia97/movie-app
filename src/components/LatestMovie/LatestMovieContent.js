import React from 'react';
import PropTypes from 'prop-types';

const Content = ({ children }) => {
  return (
    <div className="absolute flex flex-col justify-center items-center top-0 mt-3 w-full md:w-4/6 md:items-start md:pt-24 md:pl-32 lg:pt-24 lg:pl-40 xl:mt-40 xl:pl-56">
      {children}
    </div>
  );
};

Content.propTypes = {
  children: PropTypes.node,
};

Content.defaultProps = {
  children: null,
};

export default Content;
