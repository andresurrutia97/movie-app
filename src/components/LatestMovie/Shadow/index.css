.latest-movie__shadow {
  position: absolute;
  bottom: 0;
  margin-bottom: -5px;
  width: 100%;
  height: 100px;
  background-image: linear-gradient(to top, #000, transparent);
}
