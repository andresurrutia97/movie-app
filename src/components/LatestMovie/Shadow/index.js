import React from 'react';

import './index.css';

const index = () => <div className="latest-movie__shadow" />;

export default index;
