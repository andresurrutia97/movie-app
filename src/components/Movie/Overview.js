import React from 'react';
import PropTypes from 'prop-types';

const Overview = ({ overview }) => {
  return <p className="text-sm text-justify text-white">{overview}</p>;
};

Overview.propTypes = {
  overview: PropTypes.string,
};

Overview.defaultProps = {
  overview: '',
};

export default Overview;
