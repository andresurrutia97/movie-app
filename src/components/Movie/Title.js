import React from 'react';
import PropTypes from 'prop-types';

const Title = ({ title }) => {
  return (
    <h2 className="font-semibold text-xl custom-blue-text uppercase text-center md:text-left md:text-4xl">
      {title}
    </h2>
  );
};

Title.propTypes = {
  title: PropTypes.string,
};

Title.defaultProps = {
  title: '',
};

export default Title;
