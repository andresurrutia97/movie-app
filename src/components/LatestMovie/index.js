import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

import LatestMovieContent from './LatestMovieContent';
import Button from '../Ui/Button';
import Title from '../Ui/MovieTitle';
import MovieGenres from '../Ui/MovieGenres';
import Runtime from '../Ui/MovieRuntime';
import Overview from '../Ui/MovieOverview';
import ImgBackground from './Background';
import Shadow from './Shadow';
import { goToMovie } from '../../utils/navigation';

const LatestMovie = ({ movie, history }) => {
  return (
    <div className="w-full relative">
      <ImgBackground img={movie.backdrop_path} />
      <Shadow />
      <LatestMovieContent>
        <div className="flex flex-col items-center md:w-full md:items-start">
          <Title title={movie.title} />
          <MovieGenres genres={movie.genres} />
          <Runtime runtime={movie.runtime} />
          <Overview overview={movie.overview} />
        </div>
        <div className="mt-1 w-24 md:mt-12 md:w-64">
          <Button onClick={() => goToMovie(history, movie.id)}>Watch</Button>
        </div>
      </LatestMovieContent>
    </div>
  );
};

LatestMovie.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.number,
    backdrop_path: PropTypes.string,
    title: PropTypes.string,
    genres: PropTypes.array,
    runtime: PropTypes.number,
    overview: PropTypes.string,
  }).isRequired,
  history: PropTypes.any.isRequired,
};

export default withRouter(LatestMovie);
