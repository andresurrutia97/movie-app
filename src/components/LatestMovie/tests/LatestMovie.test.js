import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';

import LatestMovie from '../index';
import LatestMovieContent from '../LatestMovieContent';
import Shadow from '../Shadow';
import Background from '../Background';
import Title from '../../Ui/MovieTitle';
import Genres from '../../Ui/MovieGenres';
import Runtime from '../../Ui/MovieRuntime';
import Overview from '../../Ui/MovieOverview';
import Button from '../../Ui/Button';

describe('<LatestMovie/>', () => {
  const movie = {
    id: 123,
    backdrop_path: '/path',
    title: 'title',
    genres: [{ name: 'horror' }],
    runtime: 98,
    overview: 'overview',
  };

  const component = (
    <BrowserRouter>
      <LatestMovie movie={movie} history={[]} />
    </BrowserRouter>
  );
  let wrapper;
  beforeEach(() => {
    wrapper = mount(component);
  });

  it('should render correctly ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain one instance of <Content/>, <Shadow/>, <Background/> ', () => {
    expect(wrapper.find(LatestMovieContent)).toHaveLength(1);
    expect(wrapper.find(Shadow)).toHaveLength(1);
    expect(wrapper.find(Background)).toHaveLength(1);
  });

  it('inside <Content/> sould be an instance of <Title/>, <Genres/>, <Runtime/>, <Overview/> and <Button/>', () => {
    expect(wrapper.find(LatestMovieContent).find(Title)).toHaveLength(1);
    expect(wrapper.find(LatestMovieContent).find(Genres)).toHaveLength(1);
    expect(wrapper.find(LatestMovieContent).find(Runtime)).toHaveLength(1);
    expect(wrapper.find(LatestMovieContent).find(Overview)).toHaveLength(1);
    expect(wrapper.find(LatestMovieContent).find(Button)).toHaveLength(1);
  });
});
