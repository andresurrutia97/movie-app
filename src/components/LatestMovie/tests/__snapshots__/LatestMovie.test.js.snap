// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LatestMovie/> should render correctly  1`] = `
<div
  className="w-full relative"
>
  <img
    alt=""
    className="w-full latest-movie__img"
    src="https://image.tmdb.org/t/p/original/path"
  />
  <div
    className="latest-movie__shadow"
  />
  <div
    className="absolute flex flex-col justify-center items-center top-0 mt-3 w-full md:w-4/6 md:items-start md:pt-24 md:pl-32 lg:pt-24 lg:pl-40 xl:mt-40 xl:pl-56"
  >
    <div
      className="flex flex-col items-center md:w-full md:items-start"
    >
      <h2
        className="font-bold text-lg md:text-4xl lg:text-5xl"
      >
        title
      </h2>
      <div
        className="my-4"
      >
        horror
      </div>
      <div
        className="w-auto inline-block rounded-full custom-pink-bg py-1 px-3 my-3"
      >
        98 min
      </div>
      <p
        className="hidden md:block text-md"
      >
        overview
      </p>
    </div>
    <div
      className="mt-1 w-24 md:mt-12 md:w-64"
    >
      <button
        className="w-full focus:outline-none text-white py-2 px-4 rounded-md custom-blue-bg"
        onClick={[Function]}
        type="button"
      >
        Watch
      </button>
    </div>
  </div>
</div>
`;
